services/code_hosting/git_hub_service.rb

```diff
@@ -140,5 +374,29 @@ def set_build_status!(repo: nil, sha: nil, state: nil, target_url: nil, descript
       # if the user doesn't have write permission for the repo
       raise ex
     end
+
+    # This method shallow clones the project's repo given a branch,
+    # returns the Fastfile configuration. Always forces the clone.
+    # @param [String] branch
+    # @return [#peek_fastfile_configuration]
+    def shallow_clone(branch: nil)
+      self.class.peek_fastfile_configuration(
+        repo_url: self.project.repo_config.git_url,
+        branch: branch,
+        provider_credential: self.provider_credential,
+        path: self.project.repo_config.containing_path,
+        cache: false
+      )
+    end
+
+    protected
+
+    def repo_from_url(repo_url)
+      return repo_url.sub("https://github.com/", "")
+    end
+
+    def url_from_repo(repo)
+      return "https://github.com/" + repo unless repo.include?("https://github.com/")
```

Comment:
I think this will return `nil` 
Maybe what you want is:
`return repo.include?("https://github.com/") ? repo : "https://github.com/" + repo`